examples/EUVAC/FhirInfoCollector.py

```diff
@@ -0,0 +1,106 @@
+    TODO move reference resolving here - such as location
+    TODO make this the command to find/build the necessary input for a disclosure
+    '''
+    def execute(self, root):
+        # index contents, strip noise nodes
+        for i in root:
+            node = JsonParser.findPathSafe(i, ["resource"])
+            if node is None:
+                continue
+            nodeType = JsonParser.findPathSafe(i, ["resource", "resourceType"])
+            if nodeType is None:
+                continue
+            nodeId = JsonParser.findPath(i, ["resource", "id"])
+            if nodeId is None:
+                continue
+            if nodeType == "Immunization":
+                self._result.immunizations[nodeId] = node
+                continue
+            if nodeType == "Patient":
+                self._result.patients[nodeId] = node
+                continue
+            if nodeType == "Location":
+                self._result.locations[nodeId] = node
+                continue
+            if nodeType == "Organization":
+                self._result.organisations[nodeId] = node
+                continue
+            # TODO any practitioners in root?
+
+        # resolve references, map immunizations to patient
+        for i in self._result.immunizations.values():
+            self._resolvePatients(i)
+            self._resolvePerformers(i)
+            #TODO was patientId = JsonParser.findPathSafe(i, ["patient", "id"])
+            patientId = JsonParser.findPathSafe(i, ["patient", "reference"])
+            if patientId is None:
+                continue
+
+            patientId=patientId[8:]
+
+            if patientId not in self._result.immunizedPatients:
+                self._result.immunizedPatients[patientId] = ImmunizedPatient()
+
+            self._result.immunizedPatients[patientId].immunizations.append(JsonParser.findPath(i, ["id"]))
+
+        return self._result
+
+    def _resolvePatients(self, immunization):
+        patientId = JsonParser.findPathSafe(immunization, ["patient", "patientId"])
+        if not patientId is None and not patientId in self._result.patients.keys():
+            self._result.patients[patientId] = self.__resolveReference("Patient/" + patientId)
+
+    def _resolvePerformers(self, immunization):
+        actors = JsonParser.findPathSafe(immunization, ["performer"])
+        if actors is None:
+            return
+
+        for i in actors:
+            ref = JsonParser.findPathSafe(i, ["reference"])
+            if ref is None:
+                continue
+            if ref.startswith("Organization/") and not ref in self._result.organisations:
+                found = self.__resolveReference(ref)
+                self._result.organisations[ref] = found
+            if ref.startswith("Practitioner/") and not ref in self._result.practitioners:
+                found = self.__resolveReference(ref)
+                self._result.practitioners[ref] = found
+
+    def __resolveReference(self, item):
```

Comment:
should be removed

Reply:
About 2 hours ago I already decided you wouldn't like amount of classes involved for a POC, so I slowly merged  the builders/mappers and far less code.

I did lost unit tests with the last commit though as they worked on the POD/OO version.

That's first on the TODO list now.

Reply:
Also removed as it was dead code.